### PR TITLE
Minor clarifications and fixes

### PR DESCRIPTION
**Pull request by StachuDotNet**
- the README.md's  project structure seemed out of date
- Dark's source code is a bit ahead of this documentation (e.g. regarding `fsharp-backend` subfolders); this adjusts relevant documentation
- additional minor clarifications/nitpicks

### CODE REVIEW

**Review by pbiggar — COMMENTED** [review]
Looks great!

CI is failing with:

```
docs/terms-of-service.md:33:1 MD018/no-missing-space-atx No space after hash on atx style heading [Context: "#status channel in our Slack i..."]
```

There's also one typo to fix.

File: docs/contributing/adding-a-language-feature.md [hunk 1]
```diff
@@ -187,7 +187,7 @@ them. It does this over JSON.
 
 The F# backend has automatic JSON serializers and deserializers, using automatic
 serializers of types in
-[Api.fs](https://github.com/darklang/dark/blob/main/fsharp-backend/src/ApiServer/Api.fs).
+[Api.fs](https://github.com/darklang/dark/blob/main/fsharp-backend/src/ApiServer/Api).
```
[the remark this answers is not in the document]
Comment (StachuDotNet) [reply docs/contributing/adding-a-language-feature.md]: Good  catch!

**Comment by StachuDotNet** [comment]
Resolved formatting+linting issues; addressed typo

**Comment by pbiggar** [comment]
Thank you!

**Pull request merged**